rust-wasm-id-allocator/wasm-id-allocator/src/lib.rs

```diff
@@ -205,3 +201,225 @@ impl InteropIdRange {
         self.count
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use distributed_id_allocator::compressor::{NormalizationError, SessionTokenError};
+
+    static SESSION_ID_1: &str = "748540ca-b7c5-4c99-83ff-c1b8e02c09d6";
+
+    #[test]
+    #[should_panic]
+    fn cluster_capacity_fract() {
+        let mut compressor = IdCompressor::new(String::from(SESSION_ID_1)).unwrap();
+        _ = compressor.set_cluster_capacity(5.5 as f64);
+    }
+
+    #[test]
+    #[should_panic]
+    fn cluster_capacity_negative() {
+        let mut compressor = IdCompressor::new(String::from(SESSION_ID_1)).unwrap();
+
+        _ = compressor.set_cluster_capacity(-2 as f64);
+    }
+
+    #[test]
+    fn generate_next_id() {
+        let mut compressor = IdCompressor::new(String::from(SESSION_ID_1)).unwrap();
+
+        assert_eq!(compressor.generate_next_id(), -1 as f64);
+    }
+
+    #[test]
+    #[should_panic]
+    fn get_token_invalid_uuid() {
+        let mut compressor = IdCompressor::new(String::from(SESSION_ID_1)).unwrap();
+
+        _ = compressor.get_token(String::from("not_a_uuid")); // Errors at SessionId::from_uuid_string()
+    }
+
+    #[test]
+    fn take_next_range() {
+        let mut compressor = IdCompressor::new(String::from(SESSION_ID_1)).unwrap();
+
+        let id_1 = compressor.generate_next_id();
+        let count_expected = 5;
+        for _ in 1..count_expected {
+            compressor.generate_next_id();
+        }
+        let interop_id_range = compressor.take_next_range();
+        assert_eq!(interop_id_range.local, id_1);
+        assert_eq!(interop_id_range.count, count_expected as f64);
+    }
+
+    #[test]
+    fn take_next_range_empty() {
+        let mut compressor = IdCompressor::new(String::from(SESSION_ID_1)).unwrap();
+
+        let interop_id_range = compressor.take_next_range();
+        assert!(interop_id_range.local.is_nan());
+        assert!(interop_id_range.count.is_nan());
+    }
+
+    #[test]
+    fn finalize_range() {
+        let mut compressor = IdCompressor::new(String::from(SESSION_ID_1)).unwrap();
+
+        for _ in 0..5 {
+            compressor.generate_next_id();
+        }
+        let interop_id_range = compressor.take_next_range();
+
+        assert!(compressor
+            .finalize_range(
+                interop_id_range.token,
+                interop_id_range.local,
+                interop_id_range.count
+            )
+            .is_ok());
+    }
+
+    #[test]
+    fn normalize_to_op_space() {
+        let mut compressor = IdCompressor::new(String::from(SESSION_ID_1)).unwrap();
+
+        for _ in 0..5 {
+            compressor.generate_next_id();
+        }
+        let interop_id_range = compressor.take_next_range();
+        _ = compressor.finalize_range(
+            interop_id_range.token,
+            interop_id_range.local,
+            interop_id_range.count,
+        );
+        assert_eq!(compressor.normalize_to_op_space(-1 as f64), 0 as f64);
```

Comment:
Instead of writing tests that would fail if we changed certain constants (e.g., don't use static values like -5), let's rewrite this to capture the generated IDs in a collection. Then, loop over them and assert that normalizing them to op space and then back results in the same value; that's the property we care about, not which concrete numbers they are.